### ISSUE TITLE
Default for deserialization parameter ("missing")?

### ISSUE THREAD

**Issue by sloria** [marshmallow-code/marshmallow#115]
There was some discussion in #45 over the addition of a field parameter (called `missing`) that would specify a default value for deserialization. We decided not to add the parameter because it would add complexity to the API when there are already hooks--including `make_object` and `preprocessor`--that could meet common use cases.

Some comments from that thread, snipped for brevity:

Me: 

> I think there may be a use case for a second default parameter for missing input values, similar to the missing parameter in colander: http://colander.readthedocs.org/en/latest/null.html.

@zerodivision:

>  ...It seems to me that additional missing parameter is over-engineering. But setting default value for single field in preprocessor is bad too. In the case when there is a lot of such fields it requires repeated code for each field. So I guess that missing parameter is better for these cases.

Me:

> I'm going to hold off on adding the missing parameter for now, to avoid the possibility of adding unneeded code...This isn't to say that the parameter won't be added in future releases. A compelling use case may arise that cannot be cleanly addressed using the existing hooks. Until we identify that use case, it will not be included.

Thanks to @rastikerdar, the issue is revived. I think there may be added value in the `missing` parameter. Opening this up for discussion.

**Comment by rastikerdar** [comment]
Sometimes we need a callable missing:

``` python
afield = fields.DateTime(missing=lambda: datetime.datetime.utcnow())
```

And also needs to be aware of `only` and `exclude` list defined at schema initialization:

``` python
# for ...
    if name in self.exclude or (self.only and name not in self.only):
        continue # skip processing this missing field
```

**Comment by sloria** [comment]
I agree. Thanks for the suggestions, @rastikerdar .

**Comment by sloria** [comment]
Closed by #134
